pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -75,7 +75,13 @@ def _check_consider_iterating_dictionary(self, node: nodes.Call) -> None:
             return
         if node.func.attrname != "keys":
             return
-        if not isinstance(node.parent, (nodes.For, nodes.Comprehension)):
+        if not (
+            isinstance(node.parent, (nodes.For, nodes.Comprehension))
+            or (
+                isinstance(node.parent, nodes.Compare)
+                and isinstance(node.parent.parent, nodes.If)
+            )
```

Comment:
The brackets are redundant. `and` binds stronger than `or`.

Reply:
Always get these messed up..